src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/Program.cs

```diff
@@ -12,14 +12,32 @@ private static void Main(string[] args)
         {
             using (ServiceHost wcfservicehost = new ServiceHost(typeof(WcfService)))
             using (ServiceHost userNameService = new ServiceHost(typeof(WcfUserNameService)))
+            using (ServiceHost wcfDuplexService = new ServiceHost(typeof(WcfDuplexService)))
             {
                 wcfservicehost.Open();
                 Console.WriteLine("WcfService is ready");
                 userNameService.Open();
                 Console.WriteLine("Wcf user name service is ready");
+                wcfDuplexService.Open();
+                Console.WriteLine("WcfDuplexService is ready");
+
+                DumpEndpoints(wcfservicehost, userNameService, wcfDuplexService);
+
                 Console.WriteLine("Press ENTER to close services...");
                 Console.ReadLine();
             }
         }
+
+         static void DumpEndpoints(params ServiceHost[] hosts)
+         {
```

Comment:
@iamjasonp I don't think its possible without walking the heap or attaching something like an on open handler. I just did this to keep it simple and get an output of all the endpoints when the exe starts up.